Mathlib/Algebra/Polynomial/Factors.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2025 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import Mathlib.Algebra.EuclideanDomain.Field
+import Mathlib.Algebra.Polynomial.Roots
+
+/-!
+# Split polynomials
+
+A polynomial `f : R[X]` factors if it is a product of constant and monic linear polynomials.
+
+## Main definitions
+
+* `Polynomial.Factors f`: A predicate on a polynomial `f` saying that `f` is a product of
+  constant and monic linear polynomials.
+
+-/
+
+variable {R : Type*}
+
+namespace Polynomial
+
+section Semiring
+
+variable [Semiring R]
+
+/-- A polynomial factors if a product of constant and monic linear polynomials. -/
+def Factors (f : R[X]) : Prop := f ∈ Submonoid.closure ({C a | a : R} ∪ {X + C a | a : R})
+
+theorem factors_C (a : R) : Factors (C a) :=
+  Submonoid.mem_closure_of_mem (Set.mem_union_left _ ⟨a, rfl⟩)
+
+theorem factors_zero : Factors (0 : R[X]) := by
+  simpa using factors_C (0 : R)
+
+theorem factors_one : Factors (1 : R[X]) := by
+  simpa using factors_C (1 : R)
+
+theorem factors_X_add_C (a : R) : Factors (X + C a) :=
+  Submonoid.mem_closure_of_mem (Set.mem_union_right _ ⟨a, rfl⟩)
+
+theorem factors_X : Factors (X : R[X]) := by
+  simpa using factors_X_add_C (0 : R)
+
+protected theorem Factors.mul {f g : R[X]} (hf : Factors f) (hg : Factors g) :
```

Comment:
hmm... should we use `Factors.` in the other theorem names?

Reply:
I've used `Factors.` when it allows for dot-notation. Not sure about the benefit more generally (e.g., `Splits` doesn't really have its own namespace).

Reply:
Yeah I think the advantage faded away when I remembered about `aesop`.